firmware/thruna/PDM/src/app/states/app_initState.c

```diff
@@ -17,6 +21,11 @@ static void initStateRunOnTick1Hz(struct StateMachine *const state_machine)
 static void initStateRunOnTick100Hz(struct StateMachine *const state_machine)
 {
     app_allStates_runOnTick100Hz(state_machine);
+
+    if (App_CanRx_BMS_State_Get() == BMS_INVERTER_ON_STATE)
+    {
+        App_SharedStateMachine_SetNextState(state_machine, app_driveState_get());
+    }
```

Comment:
To make this more robust, maybe also go to drive state if the BMS is PRECHARGE or DRIVE